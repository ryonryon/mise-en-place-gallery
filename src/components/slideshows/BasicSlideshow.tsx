import styled from "@emotion/styled";
import * as React from "react";

import CustomRadioButtonGroup from "../inputs/CustomRadioButtonGroup";
import SlideshowImages from "./SlideshowImages";
import SquareIconButton from "../buttons/SquareIconButton";
import { LeftArrow, RightArrow } from "./Arrow";

interface Props {
  width: string;
  height: string;
  imgSrcs: string[];
}

const BasicSlideshow = ({ width, height, imgSrcs }: Props) => {
  const SlideshowDiv = styled.div`
    display: block;
    position: relative;
    width: ${width};
    height: ${height};
    margin: 0 auto;

    .button-group {
      position: absolute;
      top: 50%;
      width: 100%;
      display: flex;
      flex-flow: row nowrap;
      justify-content: space-between;
    }

    .radio-group {
      position: absolute;
      bottom: 1%;
      width: 100%;
      text-align: center;
    }
  `;

  const [checkedValue, setCheckedValue] = React.useState(0);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const value = event.currentTarget.value;

    if (value === "left") {
      setCheckedValue(prevValue => (prevValue === 0 ? prevValue : prevValue - 1));
    } else {
      setCheckedValue(prevValue => (prevValue === imgSrcs.length - 1 ? prevValue : prevValue + 1));
    }
  };

  const handleRadioClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const value = event.currentTarget.value;
    setCheckedValue(Number(value));
  };

  return (
    <SlideshowDiv>
      <SlideshowImages imgSrcs={imgSrcs} activeNumber={checkedValue} />
      <div className="button-group">
        <SquareIconButton onClick={handleClick} value="left" disabled={0 === checkedValue ? true : false}>
          <LeftArrow />
        </SquareIconButton>
        <SquareIconButton
          onClick={handleClick}
          value="right"
          disabled={checkedValue === imgSrcs.length - 1 ? true : false}
        >
          <RightArrow />
        </SquareIconButton>
      </div>
      <div className="radio-group">
        <CustomRadioButtonGroup
          name="slideshow"
          labels={imgSrcs.map(_ => "")}
          display="inline-block"
          onClick={handleRadioClick}
          defaultChecked={checkedValue}
        />
      </div>
    </SlideshowDiv>
  );
};

BasicSlideshow.defaultProps = {
  width: "90%",
  height: "90%"
};

export default BasicSlideshow;
