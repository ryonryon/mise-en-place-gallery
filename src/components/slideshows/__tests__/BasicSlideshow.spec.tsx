import * as React from "react";
import * as renderer from "react-test-renderer";

import BasicSlideshow from "../BasicSlideshow";

describe("Arrow test", () => {
  it("tests Arrow render correctly", () => {
    const basicSlideshow = renderer.create(<BasicSlideshow imgSrcs={["test", "test"]} />);

    expect(basicSlideshow).toMatchSnapshot();
  });
});
