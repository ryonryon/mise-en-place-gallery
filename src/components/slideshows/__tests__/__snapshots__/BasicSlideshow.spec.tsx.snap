// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow test tests Arrow render correctly 1`] = `
<div
  className="css-95n64y"
>
  <div
    className="css-tpx3lo"
  >
    <img
      alt="img-0"
      className="img "
      src="test"
    />
    <img
      alt="img-1"
      className="img img-inactive"
      src="test"
    />
  </div>
  <div
    className="button-group"
  >
    <button
      className="css-plbu26"
      disabled={true}
      onClick={[Function]}
      value="left"
    >
      <svg
        height="30px"
        viewBox="0 0 50 80"
        width="30px"
        xmlSpace="preserve"
      >
        <polyline
          fill="none"
          points="45.63,75.8 0.375,38.087 45.63,0.375"
          stroke="#FFFFFF"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
      </svg>
    </button>
    <button
      className="css-1o5go31"
      disabled={false}
      onClick={[Function]}
      value="right"
    >
      <svg
        height="30px"
        viewBox="0 0 50 80"
        width="30px"
        xmlSpace="preserve"
      >
        <polyline
          fill="none"
          points="0.375,0.375 45.63,38.087 0.375,75.8"
          stroke="#FFFFFF"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
        />
      </svg>
    </button>
  </div>
  <div
    className="radio-group"
  >
    <div
      className="radio-group css-q1v34d"
    >
      <div>
        <label
          className="checked"
        >
          <input
            defaultChecked={true}
            name="slideshow"
            onClick={[Function]}
            type="radio"
            value={0}
          />
          
        </label>
      </div>
      <div>
        <label
          className=""
        >
          <input
            defaultChecked={false}
            name="slideshow"
            onClick={[Function]}
            type="radio"
            value={1}
          />
          
        </label>
      </div>
    </div>
  </div>
</div>
`;
